ansible/roles/operator-pipeline/templates/openshift/tasks/acquire-lock.yaml

```diff
@@ -0,0 +1,69 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: acquire-lease
+spec:
+  params:
+  - name: lease-name
+    type: string
+  - name: timeout
+    type: string
+    default: 90m
+  stepTemplate:
+    env:
+      - name: TOTAL_DURATION_IN_SECONDS
+        value: ${0}
+  steps:
+    - name: create-lease
+      image: image-registry.openshift-image-registry.svc:5000/openshift/cli:latest
+      script: |
+        calculate_duration_in_seconds() {
+            if [ "${1: -1}" == "m" ]; then
+                TOTAL_DURATION_IN_SECONDS=$((${1:: -1} * 60))
+            # If it is "h", then the token is in hours
+            elif [ "${1: -1}" == "h" ]; then
+                TOTAL_DURATION_IN_SECONDS=$((${1:: -1} * 3600))
+            elif [ "${1: -1}" == "s" ]; then
+                TOTAL_DURATION_IN_SECONDS=$((${1:: -1}))
+            # Otherwise, the token is in seconds
+            else
+                TOTAL_DURATION_IN_SECONDS=${1}
+            fi
+        }
+        acquire_lease() {
+            echo "Attempting to create lease $(params.lease-name)"
+            until kubectl create -f e2e-lease.yaml
+            do
+              echo "Waiting for lease..."
+              acquiredTime=$(oc get lease $(params.lease-name) -o=jsonpath='{.spec.acquireTime}')
+              acquiredTimeInSeconds=$(date +%s --date="${acquiredTime}")
+              currentTimeinSeconds=$(date +%s)
+              elapsedSeconds=$(($currentTimeinSeconds-$acquiredTimeInSeconds))
+              if (( elapsedSeconds > TOTAL_DURATION_IN_SECONDS)); then
+                echo "Lease abandoned, deleting"
+                kubectl delete lease $(params.lease-name)
+              fi
+            done
```

Comment:
If the lease is being held by another pipelinerun, this will keep hammering the openshift api as fast as it can.
I would add a sleep of a few seconds in the loop.